docs/data/data-grid/server-side-data/ServerSideLazyLoadingErrorHandling.tsx

```diff
@@ -0,0 +1,111 @@
+import * as React from 'react';
+import {
+  DataGridPro,
+  useGridApiRef,
+  GridToolbar,
+  GridDataSource,
+  GridGetRowsParams,
+  GRID_ROOT_GROUP_ID,
+} from '@mui/x-data-grid-pro';
+import Checkbox from '@mui/material/Checkbox';
+import FormControlLabel from '@mui/material/FormControlLabel';
+import { useMockServer } from '@mui/x-data-grid-generator';
+import Alert from '@mui/material/Alert';
+import Button from '@mui/material/Button';
+
+function ErrorAlert({ onClick }: { onClick: () => void }) {
+  return (
+    <Alert
+      sx={{
+        position: 'absolute',
+        bottom: '0',
+        paddingX: 2,
+        paddingY: 1,
+        width: '100%',
+        zIndex: 10,
+      }}
+      severity="error"
+      action={
+        <Button color="inherit" size="small" onClick={onClick}>
+          Retry
+        </Button>
+      }
+    >
+      Could not fetch the data
+    </Alert>
+  );
+}
```

Comment:
```suggestion
function ErrorSnackbar(props: SnackbarProps & { onRetry: () => void }) {
  const { onRetry, ...rest } = props;
  return (
    <Snackbar {...rest}>
      <Alert
        severity="error"
        variant="filled"
        sx={{ width: '100%' }}
        action={
          <Button color="inherit" size="small" onClick={onRetry}>
            Retry
          </Button>
        }
      >
        Failed to fetch row data
      </Alert>
    </Snackbar>
  );
}
```

Reply:
yeah, I planned to change this as well, to be aligned with other demos